src/com/google/enterprise/adaptor/documentum/DocumentumAdaptor.java

```diff
@@ -903,8 +913,21 @@ protected boolean fillCollection(IDfSession dmSession,
     @Override
     protected Checkpoint pushCollection(DocIdPusher pusher)
         throws InterruptedException {
+      FeedType feedType;
```

Comment:
This logic is messy. Two things seem worth noting: 1) we never set the feed type to FULL, and 2) we never change more than once per instance. So we could just overwrite this.feedType if prevError || (queryBatchSize > 0 && prevGroups). Also, do we really care if an exception is thrown? It seems like what we really care about is just whether pushGroupDefinitions has been called before, or called with a non-empty map. (I think there may be a bug in the library if a full group feed is done with an empty map, but that's an aside.)